### PR TITLE
Quaternion from axis angle representation 

### PR DESCRIPTION
**Pull request by cjpurackal**
#### Changes

Feat: Quaternion from angle axis representation

#### Type of change

- [x] 📚  Documentation Update
- [x] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

File: kornia/geometry/quaternion.py [hunk 1]
```diff
@@ -244,6 +244,27 @@ def from_matrix(cls, matrix: Tensor) -> 'Quaternion':
         """
         return cls(rotation_matrix_to_quaternion(matrix, order=QuaternionCoeffOrder.WXYZ))
 
+    @classmethod
+    def from_axis_angle(cls, axis: Tensor, angle: Tensor) -> 'Quaternion':
```
Comment (ducha-aiki) [inline kornia/geometry/quaternion.py:248]: Why not just call `angle_axis_to_quaternion` ? Also, why 2 inputs instead of angle * axis?
 https://github.com/kornia/kornia/blob/master/kornia/geometry/conversions.py#L786
 
 Also on the first

Comment (cjpurackal) [reply kornia/geometry/quaternion.py:248]: Oops, didn't realize the conversion was already there! Yes, angle*axis seems to be a better representation!

**Comment by cjpurackal** [comment]
updated @edgarriba @ducha-aiki

File: kornia/geometry/quaternion.py [hunk 2]
```diff
@@ -244,6 +245,20 @@ def from_matrix(cls, matrix: Tensor) -> 'Quaternion':
         """
         return cls(rotation_matrix_to_quaternion(matrix, order=QuaternionCoeffOrder.WXYZ))
 
+    @classmethod
+    def from_angle_axis(cls, angle_axis: Tensor) -> 'Quaternion':
```
Comment (edgarriba) [inline kornia/geometry/quaternion.py:249]:
```suggestion
    def from_axis_angle(cls, axis_angle: Tensor) -> 'Quaternion':
```
At some point we need to update the naming for this function too in conversions, let just start from here

File: kornia/geometry/quaternion.py [hunk 3]
```diff
@@ -244,6 +245,20 @@ def from_matrix(cls, matrix: Tensor) -> 'Quaternion':
...
+        """Create a quaternion from angle-axis representation
```
Comment (edgarriba) [inline kornia/geometry/quaternion.py:250]:
```suggestion
        """Create a quaternion from angle-axis representation.
        
```

File: kornia/geometry/quaternion.py [hunk 4]
```diff
@@ -244,6 +245,20 @@ def from_matrix(cls, matrix: Tensor) -> 'Quaternion':
...
+        Args:
+            angle_axis: rotation vector of shape :math:`(B,3)`.
```
Comment (edgarriba) [inline kornia/geometry/quaternion.py:252]:
```suggestion
            angle_axis: rotation vector of shape :math:`(B,3)`.
            
```

File: kornia/geometry/quaternion.py [hunk 5]
```diff
@@ -244,6 +245,20 @@ def from_matrix(cls, matrix: Tensor) -> 'Quaternion':
...
+        Example:
+            >>> angle_axis = torch.Tensor([[1, 0, 0]])
```
Comment (edgarriba) [inline kornia/geometry/quaternion.py:254]:
```suggestion
            >>> angle_axis = torch.tensor([[1, 0, 0]])
```

**Review by edgarriba — APPROVED** [review]
LGTM. fix comments

**Comment by cjpurackal** [comment]
> LGTM. fix comments

Done
